tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/TeamMailboxMemberSetMethod.scala

```diff
@@ -0,0 +1,121 @@
+import org.apache.james.jmap.routes.SessionSupplier
+import org.apache.james.mailbox.MailboxSession
+import org.apache.james.metrics.api.MetricFactory
+import org.apache.james.user.api.UsersRepository
+import org.reactivestreams.Publisher
+import play.api.libs.json.JsObject
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+class TeamMailboxMemberSetMethod @Inject()(val teamMailboxRepository: TeamMailboxRepository,
+                                           val usersRepository: UsersRepository,
+                                           val metricFactory: MetricFactory,
+                                           val sessionTranslator: SessionTranslator,
+                                           val sessionSupplier: SessionSupplier) extends MethodRequiringAccountId[TeamMailboxMemberSetRequest] {
+
+  override val methodName: Invocation.MethodName = MethodName("TeamMailboxMember/set")
+
+  override val requiredCapabilities: Set[CapabilityIdentifier] = Set(JMAP_CORE, JMAP_MAIL, LINAGORA_TEAM_MAILBOXES)
+
+  override def getRequest(mailboxSession: MailboxSession, invocation: Invocation): Either[Exception, TeamMailboxMemberSetRequest] =
+    Serializer.deserializeSetRequest(invocation.arguments.value)
+      .asEither.left.map(ResponseSerializer.asException)
+
+  override def doProcess(capabilities: Set[CapabilityIdentifier],
+                         invocation: InvocationWithContext,
+                         mailboxSession: MailboxSession,
+                         request: TeamMailboxMemberSetRequest): Publisher[InvocationWithContext] =
+    update(mailboxSession.getUser, request)
+      .map(response => Invocation(
+        methodName = methodName,
+        arguments = Arguments(Serializer.serializeSetResponse(response).as[JsObject]),
+        methodCallId = invocation.invocation.methodCallId))
+      .map(InvocationWithContext(_, invocation.processingContext))
+
+  private def update(username: Username, request: TeamMailboxMemberSetRequest): SMono[TeamMailboxMemberSetResponse] =
+    SFlux.fromIterable(request.validatedUpdateRequest())
+      .flatMap(parsingRequestResult => update(username, parsingRequestResult))
+      .collectSeq()
+      .map(listResult => TeamMailboxMemberSetResponse.from(request.accountId, listResult))
+
+  private def update(username: Username,
+                     parsingRequestResult: ParsingRequestResult): SMono[TeamMailboxMemberSetResult] =
+    parsingRequestResult match {
+      case ParsingRequestFailure(teamMailboxNameDTO, exception) => SMono.just(TeamMailboxMemberSetResult.notUpdated(teamMailboxNameDTO, SetErrorDescription(exception.message)))
+      case ParsingRequestSuccess(teamMailbox, membersUpdateToAdd, membersUpdateToRemove) =>
+        checkIfAnyUsersDoesNotExist(membersUpdateToAdd, membersUpdateToRemove)
+          .filter(anyUsersNotExist => anyUsersNotExist)
+          .map(_ => TeamMailboxMemberSetResult.notUpdated(teamMailbox, SetErrorDescription("Some users do not exist in the system")))
```

Comment:
Listing which users do not exist would be better IMO.

Imagine updating multiple users at the same request. It is hard to know which user has the existance issue.